packages/experience-builder-sdk/src/hooks/useDetectEditorMode.tsx

```diff
@@ -42,7 +42,7 @@ export const useDetectEditorMode = ({ isClientSide = false }: UseDetectEditorMod
       //Double check if we are in editor mode by listening to postMessage events
       if (typeof window !== 'undefined' && !window.__EB__?.isEditorMode) {
         window.addEventListener('message', onMessage);
-        sendMessage(OUTGOING_EVENTS.Connected);
+        sendMessage(OUTGOING_EVENTS.Connected, undefined);
```

Comment:
Mmmh, not fully happy with this but maybe it's even good to make the payload explicit. I tried improving the types but failed so far. I guess it would only work with overloaded function types - an interesting challenge for the dev-typescript channel in Slack if we wanted this to be optional :)

Reply:
Thanks for the suggestion, I also tried a few different things but couldn't make it work, and didn't want to block this PR longer. But I will followup on dev-typescript, because I also don't like this solution :D